test/modules/aws-vpc-network-acl-integration.ts

```diff
@@ -0,0 +1,411 @@
+      cidr_block = '192.${randIPBlock}.0.0/16';
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+  it('applies the subnet change', commit());
+
+  it(
+    'confirms that subnet ACL has been replaced',
+    query(
+      `
+    SELECT * FROM subnet WHERE vpc_id=(SELECT id FROM vpc WHERE tags ->>'name' = '${prefix}-1') AND
+    cidr_block = '192.${randIPBlock}.0.0/16' AND network_acl_id=(SELECT id FROM network_acl WHERE  tags ->> 'name' = '${prefix}-2');
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+      },
+    ),
+  );
+
+  it('deletes subnet', begin());
+
+  it(
+    'deletes subnet',
+    query(
+      `  
+    DELETE FROM subnet WHERE vpc_id=(SELECT id FROM vpc WHERE tags ->>'name' = '${prefix}-1') AND cidr_block = '192.${randIPBlock}.0.0/16';
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('applies delete subnet', commit());
+
+  it('updates ACL entries', begin());
+  it(
+    'updates ACL entries',
+    query(
+      `  
+    UPDATE network_acl SET entries = '[{"CidrBlock":"10.0.0.0/0","Egress":true,"Protocol":"-1","RuleAction":"deny","RuleNumber":1},{"CidrBlock":"0.0.0.0/0","Egress":false,"Protocol":"-1","RuleAction":"deny","RuleNumber":2}]'
+    WHERE  tags ->> 'name' = '${prefix}-1';
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('applies the ACL change', commit());
+
+  it(
+    'confirms that network ACL entries have been modified',
+    query(
+      `
+    SELECT * FROM network_acl WHERE tags ->> 'name' = '${prefix}-1';
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+        expect(res[0]['entries']).toEqual(
+          JSON.parse(
+            `[{"CidrBlock": "0.0.0.0/0", "Egress": true, "Protocol": "-1", "RuleAction": "deny", "RuleNumber": 1}, {"CidrBlock": "0.0.0.0/0", "Egress": true, "Protocol": "-1", "RuleAction": "deny", "RuleNumber": 32767}, {"CidrBlock": "0.0.0.0/0", "Egress": false, "Protocol": "-1", "RuleAction": "deny", "RuleNumber": 2}, {"CidrBlock": "0.0.0.0/0", "Egress": false, "Protocol": "-1", "RuleAction": "deny", "RuleNumber": 32767}]`,
+          ),
+        );
+      },
+    ),
+  );
+
+  it('updates ACL tags', begin());
+
+  it(
+    'updates ACL tags',
+    query(
+      `  
+    UPDATE network_acl SET tags = '{"name":"${prefix}-3"}' WHERE  tags ->> 'name' = '${prefix}-1';
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('applies the ACL change', commit());
+
+  it(
+    'confirms that network ACL tags have been modified',
+    query(
+      `
+    SELECT * FROM network_acl WHERE tags ->> 'name' = '${prefix}-3';
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+      },
+    ),
+  );
+  it('deletes network ACLs', begin());
```

Comment:
before deleting, can we uninstall and install the modules and add a test to make sure the sync works as expected?